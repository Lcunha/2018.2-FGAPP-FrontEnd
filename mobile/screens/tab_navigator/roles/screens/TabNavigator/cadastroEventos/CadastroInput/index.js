import React, { Component } from "react";
import { View, StyleSheet } from "react-native";
import { Icon, Item, Input } from "native-base";

const styles = StyleSheet.create({
	container: {
		flex: 1,
		alignContent: "center",
		backgroundColor: "white"
	},
	inputContainer: {
		width: "80%",
		alignSelf: "center",
		justifyContent: "space-around",
		alignItems: "center",
		flexDirection: "row",
		marginTop: -5
	}
});

export default class CadastroInput extends Component {
	render() {
		return (
			<View style={styles.inputContainer}>
				<Icon
					type={this.props.iconType}
					name={this.props.iconName}
          style={{ fontSize: 30, marginTop: 15 }}
				/>
				<Item style={{ width: "80%" }}>
					<Input
						placeholder={this.props.placeholder}
            style={{ textAlign: "center" }}
            onChangeText={(text)=>this.props.onChangeText(text)}
            keyboardType = {this.props.keyboardType == null ? 'default' : this.props.keyboardType }
					/>
				</Item>
			</View>
		);
	}
}
