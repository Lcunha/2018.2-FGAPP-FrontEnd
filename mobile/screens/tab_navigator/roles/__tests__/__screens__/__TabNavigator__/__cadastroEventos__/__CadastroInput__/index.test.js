import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import CadastroInput from '../../../../../screens/TabNavigator/cadastroEventos/CadastroInput/index'

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer.create(< CadastroInput/>).toJSON();
  expect(tree).toMatchSnapshot();
});