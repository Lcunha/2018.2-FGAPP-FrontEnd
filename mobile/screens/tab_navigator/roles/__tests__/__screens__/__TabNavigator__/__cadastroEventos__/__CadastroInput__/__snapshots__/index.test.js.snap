// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "flexDirection": "row",
      "justifyContent": "space-around",
      "marginTop": -5,
      "width": "100%",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": 1,
        "borderColor": "#D9D5DC",
        "borderLeftWidth": 0,
        "borderRadius": undefined,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "borderWidth": undefined,
        "flexDirection": "row",
        "marginLeft": 2,
        "opacity": 1,
        "width": "80%",
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <TextInput
      allowFontScaling={true}
      defaultValue=""
      editable={true}
      keyboardType="default"
      onChangeText={[Function]}
      placeholder={undefined}
      placeholderTextColor="#575757"
      style={
        Array [
          Object {
            "color": "#000",
            "flex": 1,
            "fontSize": 17,
            "height": 50,
            "paddingLeft": 5,
            "paddingRight": 5,
            "top": 1.5,
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
      underlineColorAndroid="rgba(0,0,0,0)"
      value={undefined}
      virtual={undefined}
    />
  </View>
</View>
`;
